<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>696</width>
    <height>549</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="6">
       <widget class="QLineEdit" name="lineEdit_2">
        <property name="text">
         <string>COM3</string>
        </property>
       </widget>
      </item>
      <item row="0" column="8">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="4">
       <widget class="QLineEdit" name="lineEdit">
        <property name="text">
         <string>COM1</string>
        </property>
       </widget>
      </item>
      <item row="0" column="7">
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Port1(slave): </string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Port2(master): </string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Frames</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTableWidget" name="tableWidget">
          <column>
           <property name="text">
            <string>Slave ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Function code</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Data</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>CRC error check</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Registers values</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="2" column="0">
         <widget class="QTableWidget" name="tableWidget_3"/>
        </item>
        <item row="1" column="0">
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="1" column="2">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Register count: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Slave ID: </string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Address count: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QComboBox" name="comboBox_2">
            <property name="minimumSize">
             <size>
              <width>140</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>Input registers</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Holding registers</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_3">
            <property name="text">
             <string>10</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLineEdit" name="lineEdit_4">
            <property name="text">
             <string>10</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Data type: </string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Slave options</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="4" column="0">
         <layout class="QGridLayout" name="gridLayout_8">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="3">
           <widget class="QPushButton" name="pushButton_4">
            <property name="text">
             <string>Blocking writing</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="pushButton_5">
            <property name="text">
             <string>Modifiyng input registers</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="pushButton_6">
            <property name="text">
             <string>Modifiyng holding registers</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Add registers to modify/block: </string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QGridLayout" name="gridLayout_9">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="2">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_5">
            <property name="text">
             <string>10</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_8">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Number of entries: </string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="3">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_7">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Slave ID: </string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox_3">
            <property name="minimumSize">
             <size>
              <width>270</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QPushButton" name="pushButton_3">
            <property name="text">
             <string>Refresh</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="0">
         <layout class="QGridLayout" name="gridLayout_10">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="1">
           <widget class="QTableWidget" name="tableWidget_4"/>
          </item>
          <item row="1" column="0">
           <widget class="QTableWidget" name="tableWidget_2"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_10">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Holding Registers parameters</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Input Registers parameters</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>696</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMode">
    <property name="title">
     <string>Mode</string>
    </property>
    <addaction name="action2_ports"/>
    <addaction name="action1_port"/>
   </widget>
   <widget class="QMenu" name="menuPort_settings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionPort_settings"/>
    <addaction name="separator"/>
    <addaction name="actionTask_settings"/>
   </widget>
   <addaction name="menuMode"/>
   <addaction name="menuPort_settings"/>
  </widget>
  <action name="action2_ports">
   <property name="text">
    <string>2 ports</string>
   </property>
  </action>
  <action name="action1_port">
   <property name="text">
    <string>1 port</string>
   </property>
  </action>
  <action name="actionTurn_on">
   <property name="text">
    <string>Turn on</string>
   </property>
  </action>
  <action name="actionTurn_off">
   <property name="text">
    <string>Turn off</string>
   </property>
  </action>
  <action name="actionPort_settings">
   <property name="text">
    <string>Port settings</string>
   </property>
  </action>
  <action name="actionTask_settings">
   <property name="text">
    <string>Task settings</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
